### ISSUE TITLE
Allow modifying the base url of Groq

### ISSUE THREAD

**Issue by sideef5ect** [langgenius/dify#12185]
### Self Checks

- [X] I have searched for existing issues [search for existing issues](https://github.com/langgenius/dify/issues), including closed ones.
- [X] I confirm that I am using English to submit this report (我已阅读并同意 [Language Policy](https://github.com/langgenius/dify/issues/1542)).
- [X] [FOR CHINESE USERS] 请务必使用英文提交 Issue，否则会被关闭。谢谢！:）
- [X] Please do not modify this template :) and fill in all the required fields.

### 1. Is this request related to a challenge you're experiencing? Tell me about your story.

- Dify allows users to add a Groq API token to use Groq-provided models.
- The current implementation only allows setting the API token.
- Some server environments hosting the Dify backend may have restrictions preventing direct access to the Groq API service.

Proposed Enhancement:
1. Modify the Groq model provider configuration page to include an option for setting a custom API base URL, similar to the OpenAI configuration form.
2. This addition will enable users to specify a proxy server for accessing Groq services when direct access is not possible due to network restrictions.

### 3. Can you help us with this feature?

- [ ] I am interested in contributing to this feature.

**Comment by vilisper** [comment]
Agreed, that would indeed make it more user-friendly. However, there are also workaround options, such as using the OpenAI API-compatible approach, which Groq also supports.

**Comment by sideef5ect** [comment]
Other model provider might also need this option
like google gemini, the also have region restriction
<img width="731" alt="image" src="https://github.com/user-attachments/assets/e2eadafb-a78e-4edf-ae96-e620f1740a0e" />
